internal/service/route53/vpc_association_authorization.go

```diff
@@ -97,7 +104,7 @@ func resourceVPCAssociationAuthorizationRead(ctx context.Context, d *schema.Reso
 		return sdkdiag.AppendFromErr(diags, err)
 	}
 
-	vpc, err := findVPCAssociationAuthorizationByTwoPartKey(ctx, conn, zoneID, vpcID)
+	vpc, err := findVPCAssociationAuthorizationByTwoPartKey(ctx, conn, zoneID, vpcID, d.Timeout(schema.TimeoutRead))
```

Comment:
Did you consider putting the retry loop out here rather than deeper in the implementation?

Reply:
Fair point. My original thinking was to capture the `InvalidPaginationToken` as close to the original API call as possible. 

However, retrying in the read method avoids having to pass timeouts all the way down the finder call stack and makes the retry logic easier to see. I'll make the change.